src/QsCompiler/LlvmBindings/BitcodeModule.cs

```diff
@@ -380,6 +380,66 @@ public IrFunction CreateFunction(string name, IFunctionType signature)
             return Value.FromHandle<IrFunction>(valueRef)!;
         }
 
+        /// <summary>Creates a Function definition with Debug information</summary>
+        /// <param name="scope">Containing scope for the function</param>
+        /// <param name="name">Name of the function in source language form</param>
+        /// <param name="mangledName">Mangled linker visible name of the function (may be same as <paramref name="name"/> if mangling not required by source language</param>
+        /// <param name="file">File containing the function definition</param>
+        /// <param name="linePosition">1-based <see cref="DebugPosition"/> with the line number of the function definition</param>
+        /// <param name="signature">LLVM Function type for the signature of the function</param>
+        /// <param name="isLocalToUnit">Flag to indicate if this function is local to the compilation unit</param>
+        /// <param name="isDefinition">Flag to indicate if this is a definition</param>
+        /// <param name="scopeLinePosition">1-based <see cref="DebugPosition"/> with the first line of the function's outermost scope, this may not be the same as the first line of the function definition due to source formatting</param>
+        /// <param name="debugFlags">Additional flags describing this function</param>
+        /// <param name="isOptimized">Flag to indicate if this function is optimized</param>
+        /// <param name="dIBuilder"><see cref="DebugInfoBuilder"/>to use when creating debug info</param>
+        /// <returns>Function described by the arguments</returns>
+        public IrFunction CreateFunction(
```

Comment:
Not blocking for this PR, but for clarity this function could be named `CreateFunctionWithDebugInfo`.